src/platform/psp2/psp2-context.c

```diff
@@ -588,15 +584,27 @@ void _drawTex(vita2d_texture* t, unsigned width, unsigned height, bool faded, bo
 }
 
 void mPSP2Swap(struct mGUIRunner* runner) {
-	currentTex = (currentTex + 1) & 3;
-	runner->core->setVideoBuffer(runner->core, vita2d_texture_get_datap(tex[currentTex]), 256);
+	bool frameAvailable;
+	if (runner->core->platform(runner->core) == PLATFORM_GBA) {
+		struct GBA* gba = runner->core->board;
+		frameAvailable = gba->video.frameskipCounter <= 0;
```

Comment:
Interesting approach. However, I don't like the idea of having to cast to a specific core to grab the frameskip counter. It's not generic, and I want to *reduce* the number of things peering into the cores, not increase. Not sure if there's a better approach however.